### ISSUE TITLE
Disconnect stomp socket repeatedly though there has internet connection

### ISSUE THREAD

**Issue by IhChowdhury** [stomp-js/stompjs#183]
I am trying to make an chat application. I am facing problem that After connecting stomp It disconnect though there has internet connection in my pc. Than I tried several steps to reconnect again:
1. I have tried  below code from your documentation to reconnect it. But if I use this code it could not connect ever. So I leave this steps.

var client = Stomp.over(function(){
                                   return new SockJS(url);
                                });

        // Add the following if you need automatic reconnect (delay is in milli seconds)
        client.reconnect_delay = 5000; 
2. Then I tried to reconnect stomp again using below code . This method worked but it repeatedly disconnect with my server.

function initialize() {
	var socket = new SockJS('/service-sockect');
	stompClient = Stomp.over(socket);
	stompClient.connect({}, function (frame) {
		successCallBack();
	}, function (message) {
		console.log("Disconnected");
		showNotice(1,"Server Connection Lost. Reconnecting...",'green');
		reconnect('/service-sockect');

	});
};

function reconnect(socketUrl) {
	var connected = false;
	let reconInv = setInterval(() => {
	    ws = new WebSocket(socketUrl);
	    stompClient = webstomp.over(ws);
	    stompClient.connect({}, function (frame) {
	    	connected = true;
	    	successCallback();
	    }, function (message) {
	    	showNotice(1,"Server Connection Lost. Reconnecting...",'green');
	    	if(connected){
	    		reconnect(socketUrl);
	    	}	      
	    });
	  }, 1000);

};

** Now give me your opinion how can solve this issue.

**Comment by kum-deepak** [comment]
Could not understand your code. I guess you are trying a create an new application. The following resources should help you:

- https://stomp-js.github.io/guide/stompjs/2018/06/29/using-stompjs-v5.html
- https://github.com/stomp-js/samples/ - see https://github.com/stomp-js/samples/blob/master/stompjs/chat/chat.html - it has a basic chat application.
- Checkout other articles at https://stomp-js.github.io/ as you need.

**Comment by IhChowdhury** [comment]
Yeah, I am trying make an chat application using spring boot framework. I successfully connect socket by using `Stomp.over(ws) ` but it couldn't reconnect when socket is disconnected. Then I applied `var client = Stomp.over(function(){
return new SockJS(url);
});`  
 After creating client stomp using this code then my socket connection can not established. I have found this code in one of stomp documentation.
